### PR TITLE
Rename experimental config params related to mcache

### PR DESCRIPTION
**Pull request by alexttx**
Signed-off-by: Alex Tomlinson <4172734+alexttx@users.noreply.github.com>

### CODE REVIEW

**Review by tristan957 — APPROVED** [review]

File: lib/include/hse_ikvdb/kvs_rparams.h [hunk 1]
```diff
@@ -57,10 +57,10 @@ struct kvs_rparams {
...
     bool     cn_cursor_kra;
 
-    uint8_t  cn_mcache_kra_params;
-    uint8_t  cn_mcache_vra_params;
-    uint8_t  cn_mcache_wbt;
-    uint32_t cn_mcache_vmax;
+    uint8_t  cn_mmap_kra_params;
+    uint8_t  cn_mmap_vra_params;
+    uint8_t  cn_mmap_wbt;
+    uint32_t cn_mmap_vmax;
```
Comment (alexttx) [inline lib/include/hse_ikvdb/kvs_rparams.h:63]: An upcoming PR will remove KVDB's use of mcache maps.  These parameters should probably be renamed.   Thoughts on new names?  They determine how HSE uses `madvise` on memory mapped mblocks.

File: lib/kvdb/kvdb_pfxlock.c [hunk 2]
```diff
@@ -63,8 +63,8 @@ struct kvdb_pfxlock_tree {
     struct rb_root             kplt_root HSE_L1X_ALIGNED;
     struct kvdb_pfxlock_entry *kplt_ecache;
     volatile uint              kplt_entry_cnt;
-    uint                       kplt_mcache_cnt;
-    struct kvdb_pfxlock_entry *kplt_mcache;
+    uint                       kplt_cache_cnt;
+    struct kvdb_pfxlock_entry *kplt_cache;
```
Comment (alexttx) [inline lib/kvdb/kvdb_pfxlock.c:67]: This got caught up in the rename even though it's not "mcache".  The name "cache" seemed as appropriate as "mcache", and less confusing in the short term.  I saw no clear description in the prefix lock code of what the "m" stands for.

File: lib/cn/kvset.c [hunk 3]
```diff
@@ -749,11 +749,11 @@ kvset_open2(
             if (!km->km_restored) {
                 kvset_madvise_hblk(ks, MADV_WILLNEED, true);
 
-                if (ra_willneed(rp->cn_mcache_kra_params)) {
+                if (ra_willneed(rp->cn_mmap_kra_params)) {
                     kvset_madvise_kblks(ks, MADV_WILLNEED, true, true);
                 }
 
-                if (ra_willneed(rp->cn_mcache_vra_params)) {
+                if (ra_willneed(rp->cn_mmap_vra_params)) {
```
Comment (alexttx) [inline lib/cn/kvset.c:756]: `ra_willneed` is defined as:
```
#define ra_willneed(_ra)    ((_ra) & 0x01u)
```
Seems we can change param name to `cn_mmap_vra` and eliminate `ra_willneed`.

File: lib/include/hse_ikvdb/kvs_rparams.h (hunk 1 above)
Comment (alexttx) [reply lib/include/hse_ikvdb/kvs_rparams.h:63]: Would like feedback from SQL on impact of this change on automation scripts.

File: lib/cn/kvset.c (hunk 3 above)
Comment (beckerg) [reply lib/cn/kvset.c:756]: It used to contain a lot of stuff (for level based cached), but no longer, so yeah, change the name.
We should probably ignore these settings if the kblocks/vblocks are larger than RAM, nothing good can happen.  I could see maybe enabling them on a small, hot kvs, but used indiscriminantly can cause a world of hurt.

**Review by beckerg — APPROVED** [review]

File: lib/kvs/kvs_rparams.c [hunk 4]
```diff
@@ -522,12 +522,12 @@ static const struct param_spec pspecs[] = {
         },
     },
     {
-        .ps_name = "cn_mcache_vra_params",
+        .ps_name = "cn_mmap_vra_params",
```
[the remark this answers is not in the document]
Comment (alexttx) [reply lib/kvs/kvs_rparams.c:525]: This PR isn' final.  Sorry.  I will take it down and put up a new one next week.

**Pull request closed without merging**